### ISSUE TITLE
Implement repo and access federation in the labs for Artifactory

### ISSUE THREAD

**Issue by caggles** [BCDevOps/developer-experience#2827]
As part of the migration to Gold, I would like to implement repo federation and access federation to link up the Silver and Gold instances of Artifactory. Once this is finished and Gold is synced, it should be quite straightforward to gracefully move from Silver to Gold and then shut down the Silver instance. We should test this process in the lab first.

Definition of Done:

- [x] Determine the JFrog-recommended process for gracefully turning on repo and access federation
- [x] Link up the KLAB and CLAB instances of Artifactory (which will probably involve turning on Mission Control)
- [x] Test the implementation of repo and access federation across the labs

**Comment by caggles** [comment]
Project data doesn't get federated through this process, so in addition to setting up repo and access federation, we need to sync project-related data between the two instances.

- Archeobot will create new projects in both instances, and needs to add the appropriate user accounts to both projects. This should be fairly easy to do - just add the DR instance as a target to Archeobot's project-related tasks. Note that service accounts do not need to target DR, as access federation will cover that.
- A cronjob will collect project auth data from the main instance and apply it to the DR instance every X minutes. We can potentially adapt the scripts written by Helen for the SSO migration.
- All new repos created in a project must also be created in the DR instance and federated. I don't know exactly what this process should look like yet or how to automate it. This will need to be investigated.

The cronjob means that, should Artifactory fail over to the DR instance, any project privilege changes will not be applied back to the main instance when it fails back. Make sure documentation is clear about this. Also, make sure the cronjob fails gracefully if it can't connect to the main Artifactory instance (in that situation, it should exit without doing anything).

**Comment by caggles** [comment]
Testing how to federate all repos:

If you make a federated repo in a project, the paired repo in the DR instance isn't a member of that project. If you convert a local repo in a project to a federated repo, the new federated repo is no longer assigned to that project.

Hopefully JFrog support has a way to keep that project membership across both instances when creating a federated repository. If they don't, then a possible solution might be to have our users create local repos in the main instance, and then we use a cronjob to convert it to a federated repo and then re-assign both repos back to their originating project. That's very annoying and I hope there's an easier way to do that.

**Comment by caggles** [comment]
Waiting on an architecture meeting with JFrog - our new technical contact is putting together recommendations for how we should approach this, since it's becoming more complicated than I had hoped.

**Comment by caggles** [comment]
Looks like the primary option for us to continue with my original plan, despite how complicated it has turned out to be. JFrog is still looking into ways to let Project Admins create federated repos, but I don't think it's likely.

In other news, a follow-up task (after DR is finished) should be created to make Artifactory and Archeobot more "gitops-y"
